src/WebJobs.Extensions.DurableTask/LocalHttpListener.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.Mvc.WebApiCompatShim;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
+{
+    internal class LocalHttpListener : IDisposable
+    {
+        private readonly DurableTaskExtension extension;
+        private readonly IWebHost localWebHost;
+        private readonly Func<HttpRequestMessage, Task<HttpResponseMessage>> handler;
+
+        public LocalHttpListener(
+            DurableTaskExtension extension,
+            Uri listenUri,
+            Func<HttpRequestMessage, Task<HttpResponseMessage>> handler)
+        {
+            this.extension = extension ?? throw new ArgumentNullException(nameof(extension));
+            this.handler = handler ?? throw new ArgumentNullException(nameof(handler));
+            this.localWebHost = this.CreateWebHost(listenUri);
+        }
+
+        public bool IsListening { get; private set; }
+
+        public void Dispose() => this.localWebHost.Dispose();
+
+        public async Task StartAsync()
+        {
+            if (this.IsListening == true)
+            {
+                throw new InvalidOperationException("The local HTTP listener has already started.");
+            }
+
+#if !FUNCTIONS_V1
```

Comment:
We no longer need conditional compilation here now that `this.localWebHost() is a `NoOpWebHost`.

Reply:
It's still needed because `IWebHost.StartAsync()` and `IWebHost.StopAsync()` are not defined in `net471`.

Reply:
That is annoying. We should probably remove the Start/StopAsync() methods on `NoOpWebHost`, since they are functionally useless due to the differences in IWebHost in netstandard2.0 and net471.